java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java

```diff
@@ -24,82 +24,60 @@
 import java.sql.Driver;
 import java.sql.DriverManager;
 import java.sql.SQLException;
+import java.util.Collection;
 import java.util.Properties;
 
+import org.apache.arrow.driver.jdbc.adhoc.MockFlightSqlProducer;
+import org.apache.arrow.driver.jdbc.authentication.UserPasswordAuthentication;
 import org.apache.arrow.driver.jdbc.client.ArrowFlightSqlClientHandler;
 import org.apache.arrow.driver.jdbc.utils.ArrowFlightConnectionConfigImpl.ArrowFlightConnectionProperty;
-import org.apache.arrow.driver.jdbc.utils.FlightTestUtils;
-import org.apache.arrow.flight.CallStatus;
-import org.apache.arrow.flight.FlightProducer;
-import org.apache.arrow.flight.FlightServer;
-import org.apache.arrow.flight.auth2.BasicCallHeaderAuthenticator;
-import org.apache.arrow.flight.auth2.CallHeaderAuthenticator;
-import org.apache.arrow.flight.auth2.GeneratedBearerTokenAuthenticator;
 import org.apache.arrow.memory.BufferAllocator;
 import org.apache.arrow.memory.RootAllocator;
 import org.apache.arrow.util.AutoCloseables;
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.ClassRule;
 import org.junit.Test;
 
-import com.google.common.base.Strings;
-
 /**
  * Tests for {@link Connection}.
- * TODO Update to use {@link FlightServerTestRule} instead of {@link FlightTestUtils}
  */
 public class ConnectionTest {
 
-  private FlightServer server;
-  private FlightServer server2;
-  private String serverUrl;
-  private String serverUrl2;
+  @ClassRule
+  public static final FlightServerTestRule FLIGHT_SERVER_TEST_RULE;
+  private static final MockFlightSqlProducer PRODUCER = new MockFlightSqlProducer();
+
+  static {
+    UserPasswordAuthentication authentication =
+        new UserPasswordAuthentication.Builder().user("user1", "pass1").user("user2", "pass2")
+            .build();
+
+    FLIGHT_SERVER_TEST_RULE = new FlightServerTestRule.Builder().host("localhost").randomPort()
+        .authentication(authentication).producer(PRODUCER).build();
+  }
+
+  private final String keyStorePath = this.getClass().getResource("/keys/keyStore.jks")
+      .getPath();
+  private final String noCertificateKeyStorePath =
+      this.getClass().getResource("/keys/noCertificate.jks")
+          .getPath();
+  private final String keyStorePass = "flight";
```

Comment:
Why setting this parameter there? By the name of the class, this just test a normal connection without tls

Reply:
fixed

Reply:
fixed